akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -412,6 +412,20 @@ object Source {
     })
 
   /**
+   * Combine the elements of multiple streams into a stream of sequences.
+   */
+  def zipN[T](sources: Seq[Source[T, _]]) = zipWithN((s: Seq[T]) ⇒ s)(sources)
+
+  /*
+   * Combine the elements of multiple streams into a stream of sequences using a combiner function.
+   */
+  def zipWithN[T, O](zipper: Seq[T] ⇒ O)(sources: Seq[Source[T, _]]): Source[O, NotUsed] = sources match {
+    case Seq()          ⇒ empty[O]
+    case Seq(source)    ⇒ source.map(t ⇒ zipper(Seq(t))).mapMaterializedValue(_ ⇒ NotUsed)
+    case s1 +: s2 +: ss ⇒ combine(s1, s2, ss: _*)(ZipWithN(zipper))
```

Comment:
It looks that `combine` is ignoring the name of the result of the strategy. I am going to add a name attribute here too.